flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -216,6 +217,9 @@ protected FlinkConfigBuilder applyJobOrSessionSpec() throws URISyntaxException {
                 effectiveConfig.set(
                         CoreOptions.DEFAULT_PARALLELISM, spec.getJob().getParallelism());
             }
+            if (spec.getJob().isAllowNonRestoredState()) {
+                effectiveConfig.set(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE, true);
```

Comment:
I think we could make the `allowNonRestoredState` Boolean instead of boolean and set it into the config regardless of the value if it's not null. 

With the current logic if the user defines true in the flinkConfiguration and sets it to false in the jobspec we would not use it.

Reply:
we could also add a test for this to make sure the spec takes precedence

Reply:
Good point, fixed